bundles/org.eclipse.cdt.lsp.editor.ui.test/src/org/eclipse/cdt/lsp/editor/ui/test/preference/LspEditorPreferencesTesterTest.java

```diff
@@ -0,0 +1,144 @@
+package org.eclipse.cdt.lsp.editor.ui.test.preference;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.eclipse.cdt.lsp.LspPlugin;
+import org.eclipse.cdt.lsp.editor.ui.test.TestUtils;
+import org.eclipse.cdt.lsp.server.ICLanguageServerProvider;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LspEditorPreferencesTesterTest {
+	private static final String LSP_CEDITOR_ID = "org.eclipse.cdt.lsp.CEditor"; //$NON-NLS-1$
+	private static final String C_EDITOR_ID = "org.eclipse.cdt.ui.editor.CEditor"; //$NON-NLS-1$
+	private static final String FILE_CONTENT = "// sample file content";
+	private static final String MAIN_CPP = "main.cpp";
+	private static final String EXTERNAL_HEADER_HPP = "ExternalHeader.hpp";
+	private IProject project;
+	
+	@Before
+	public void setUp() throws CoreException {
+		project = TestUtils.createCProject("LspCEditorTest"+System.currentTimeMillis());
+	}
+	
+	/**
+	 * Tests whether the LS enable returns true on a resource whose project has "Prefer C/C++ Editor (LSP)" enabled.
+	 */
+	@Test
+	public void testLsEnableByUriTest_WITH_LsEditorPreferred() throws CoreException {
+		final String fileName = "main.cpp";
+		//GIVEN is a project with ENABLED "Prefer C/C++ Editor (LSP)" in the preferences:
+		TestUtils.setLspPreferred(project, true);
```

Comment:
Similar to my above comment, the defaults should be restored at the end of the test.

Reply:
Okay, but is this necessary when the project gets deleted after every test by?:
```java
	@After
	public void cleanUp() throws CoreException {
		TestUtils.deleteProject(project);
	}
```

Reply:
No - sorry it is irrelevant in this case. I hadn't paid enough attention to realize that `setLspPreferred` was acting on `ProjectScope`